### ISSUE TITLE
recent firmwares very actively print garbage on the console

### ISSUE THREAD

**Issue by Habbie** [nodemcu/nodemcu-firmware#1474]
### Expected behavior
Powering the device quickly gives me a Lua prompt.

### Actual behavior
Console is filled with garbage, speed apparently mostly dependent on the baudrate I pick. Blue LED on Nodemcu board also blinks actively.

### NodeMCU version
Last good: [8866297c6d7cc35afd25b0f0f666ef90af040300] Remove buffer:write() as ws2812.write() handles buffers (#1408)
First bad: [84c005e85c78b4382e68633ab61e2f4928c838f5] Update to SDK 1.5.4.1 (#1390)

Current dev and master heads are also bad.

In short, the 1.5.4 upgrade commit breaks building for me; the 1.5.4.1 upgrade commit unbreaks building but yields useless firmware.

### Hardware
Bisect done on NodeMCU board. Last good and first known bad commits also tested on Wemos D1 Mini with the same result.

**Comment by devsaurus** [comment]
Please see #1468 et.al. for the correct firnware upgrade procedure.
To build 1.5.4 you need to remove the 1.5.4.1 zip files from `cache` folder. Makefile was too greedy with wildcard matching back then.

**Comment by Habbie** [comment]
I did an `esptool erase_flash` before every `write_flash`, it was my understanding this is a correct upgrade procedure. Emptying the cache helps for the unbuildables, trying those now.

**Comment by Habbie** [comment]
Re-bisected with working 1.5.4 builds, updated original text.

**Comment by VSpectrum** [comment]
Following #1468 , I flashed the esp_init_data_default.bin. Although I was using an ESP-12E (4 MB) flashing the init_data to 0x3fc000 resulted in the same problem (endless blinking after reset and garbage output). However flashing to address 0x7c000 seemed to work for me:
 
esptool.py --port /dev/ttyUSB0 write_flash 0x00000 nodemcu-master-10-modules-2016-08-25-16-25-29-float.bin 0x7c000 esp_init_data_default.bin 

Is the note made concerning the addresses to flash to in http://nodemcu.readthedocs.io/en/master/en/flash/ confirmed?

**Comment by devsaurus** [comment]
Yes, the addresses are working, given you configured the flash correctly.
Please also specify the command line parameters `-fm` and `-fs` as mentioned there.

**Comment by VSpectrum** [comment]
```sh 
$ esptool.py --port /dev/ttyUSB0 write_flash 0x00000 nodemcu-master-10-modules-2016-08-25-16-25-29-float.bin 0x3fc000 esp_init_data_default.bin -fm dio -fs 32m 
```
Adding and specifying the flashmode and flashsize flags worked for me, thanks.

**Comment by Habbie** [comment]
`-fm dio -fs 32m` also work on my NodeMCU. Apparently these things have become stricter with SDK 1.5.4.1. Thank you for your patience, @devsaurus!

**Comment by rohit-gandhe** [comment]
Where do I get esp_init_data_default.bin? I don't see it in my bin dir after the build.

**Comment by rohit-gandhe** [comment]
Never mind, downloaded as part of the SDK
